cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -214,7 +215,58 @@ fn lower_insn_to_regs<'a>(ctx: Ctx<'a>, inst: IRInst) {
                 unimplemented!("unimplemented lowering for opcode {:?}", op);
             }
         }
-
+        Opcode::Fcopysign => {
+            let dst = output_to_reg(ctx, inst, 0);
+            let lhs = input_to_reg(ctx, inst, 0);
+            let rhs = input_to_reg(ctx, inst, 1);
+            let is64 = flt_ty_is_64(ty.unwrap());
+            if !is64 {
+                let w64 = 0x8000_0000;
+                let tregI64 = ctx.alloc_tmp(RegClass::I64, I32);
+                let tregV128 = ctx.alloc_tmp(RegClass::V128, F64);
+                let t2regV128 = ctx.alloc_tmp(RegClass::V128, F64);
+                let t3regV128 = ctx.alloc_tmp(RegClass::V128, F64);
```

Comment:
A few questions:

- should this be F32 and not F64, since they only use the lower 32 bits?
- what do you think about naming these `scratch_gpr`, `scratch_fp1` and so on, or simply `tmp_gpr`, `tmp_1` etc? Having the register class's name in the variable's name feels a bit misleading: one might think that a given register is a full SIMD one, while it's not quite true.
- could one of these V128 temporaries be avoided if we'd use the destination operand as a temporary?